addons/payment_adyen/static/src/js/payment_form.js

```diff
@@ -54,10 +54,12 @@ odoo.define('payment_adyen.payment_form', require => {
          * @return {undefined}
          */
         _dropinOnError: function (error) {
-            this._displayError(
-                _t("Incorrect Payment Details"),
-                _t("Please verify your payment details.")
-            );
+            if (! this.$('div[name="o_payment_error"]')){
```

Comment:
A quick comment can prevent accidental removals of this fix and helps the reader understands something that is not obvious.
```suggestion
            if (! this.$('div[name="o_payment_error"]')){  // Don't replace a specific server error.
```